packages/legend-extension-dsl-serializer/src/models/protocols/pure/DSLSerializer_PureProtocolProcessorPlugin.ts

```diff
@@ -0,0 +1,305 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import packageJson from '../../../../package.json';
+import { V1_Binding } from './v1/model/packageableElements/store/V1_Binding';
+import { V1_SchemaSet } from './v1/model/packageableElements/schemaSet/V1_SchemaSet';
+import { V1_ExternalFormatConnection } from './v1/model/packageableElements/connection/V1_ExternalFormatConnection';
+import type { PlainObject } from '@finos/legend-shared';
+import { assertType, guaranteeNonNullable } from '@finos/legend-shared';
+import { deserialize, serialize } from 'serializr';
+import {
+  V1_bindingModelSchema,
+  V1_BINDING_ELEMENT_PROTOCOL_TYPE,
+  V1_schemaSetModelSchema,
+  V1_SCHEMA_SET_ELEMENT_PROTOCOL_TYPE,
+  V1_externalFormatConnectionModelSchema,
+  V1_EXTERNAL_FORMAT_CONNECTION_ELEMENT_PROTOCOL_TYPE,
+} from './v1/transformation/pureProtocol/V1_DSLSerializer_ProtocolHelper';
+import {
+  getBinding,
+  getSchemaSet,
+} from '../../../graphManager/DSLSerializer_GraphManagerHelper';
+import { V1_resolveBinding } from '../pure/v1/transformation/pureGraph/V1_DSLSerializer_GraphBuilderHelper';
+import { Binding } from '../../metamodels/pure/model/packageableElements/store/Binding';
+import { SchemaSet } from '../../metamodels/pure/model/packageableElements/schemaSet/SchemaSet';
+import { ExternalFormatConnection } from '../../metamodels/pure/model/packageableElements/connection/ExternalFormatConnection';
+import type {
+  GraphPluginManager,
+  PackageableElement,
+  Connection,
+  V1_ElementProtocolClassifierPathGetter,
+  V1_ElementProtocolDeserializer,
+  V1_ElementProtocolSerializer,
+  V1_ElementTransformer,
+  V1_ConnectionBuilder,
+  V1_ConnectionTransformer,
+  V1_ConnectionProtocolSerializer,
+  V1_ConnectionProtocolDeserializer,
+  V1_GraphBuilderContext,
+  V1_GraphTransformerContext,
+  V1_Connection,
+  V1_PackageableElement,
+  PackageableElementReference,
+  Store,
+} from '@finos/legend-graph';
+import {
+  V1_transformElementReference,
+  DSLMapping_PureProtocolProcessorPlugin_Extension,
+  V1_ElementBuilder,
+  V1_initPackageableElement,
+} from '@finos/legend-graph';
+
+const BINDING_ELEMENT_CLASSIFIER_PATH =
+  'meta::external::shared::format::binding::Binding';
+const SCHEMA_SET_ELEMENT_CLASSIFIER_PATH =
+  'meta::external::shared::format::metamodel::SchemaSet';
+
+export class DSLSerializer_PureProtocolProcessorPlugin extends DSLMapping_PureProtocolProcessorPlugin_Extension {
+  constructor() {
+    super(
+      packageJson.extensions.pureProtocolProcessorPlugin,
+      packageJson.version,
+    );
+  }
+
+  install(pluginManager: GraphPluginManager): void {
+    pluginManager.registerPureProtocolProcessorPlugin(this);
+  }
+
+  override V1_getExtraElementBuilders(): V1_ElementBuilder<V1_PackageableElement>[] {
+    return [
+      new V1_ElementBuilder<V1_Binding>({
+        elementClassName: 'Binding',
```

Comment:
I'm kind of tempted to just use `Binding.name` here instead

Reply:
Do you mean replcing 'Binding' with 'Binding.name' ?

Reply:
yes, the name of the class `Binding` so we don't need to hard code with a `string` like this